presto-main/src/main/java/com/facebook/presto/operator/PartitionedOutputOperator.java

```diff
@@ -229,10 +231,8 @@ public PartitionedOutputOperator(
                 maxMemory);
 
         operatorContext.setInfoSupplier(this::getInfo);
-        // TODO: We should try to make this more accurate
-        // Recalculating the retained size of all the PageBuilders is somewhat expensive,
-        // so we only do it once here rather than in addInput(), and assume that the size will be constant.
-        operatorContext.newLocalSystemMemoryContext().setBytes(this.partitionFunction.getRetainedSizeInBytes());
+        this.systemMemoryContext = operatorContext.newLocalSystemMemoryContext();
+        this.systemMemoryContext.setBytes(this.partitionFunction.getSizeInBytes());
```

Comment:
I think it might worth caching this value (i.e., the first time we call `getSizeInBytes`, or maybe `getRetainedSizeInBytes` to be precise) somewhere as the instance size for all the page builders. So within `public long getSizeInBytes()`, we can add this instance size together with all other `sizeInBytes` as the estimated `retainedSizeInBytes`.

Reply:
Yep, good idea. We can use it to approximate the object overhead of the page builders.